bftengine/src/bftengine/ReadOnlyReplica.cpp

```diff
@@ -128,4 +137,81 @@ void ReadOnlyReplica::onMessage<CheckpointMsg>(CheckpointMsg *msg) {
   }
 }
 
+template <>
+void ReadOnlyReplica::onMessage<ClientRequestMsg>(ClientRequestMsg *m) {
+  const NodeIdType senderId = m->senderId();
+  const NodeIdType clientId = m->clientProxyId();
+  const bool reconfig_flag = (m->flags() & MsgFlag::RECONFIG_FLAG) != 0;
+  const ReqId reqSeqNum = m->requestSeqNum();
+  const uint8_t flags = m->flags();
+
+  SCOPED_MDC_CID(m->getCid());
+  LOG_DEBUG(MSGS, KVLOG(clientId, reqSeqNum, senderId) << " flags: " << std::bitset<8>(flags));
+
+  const auto &span_context = m->spanContext<std::remove_pointer<decltype(m)>::type>();
+  auto span = concordUtils::startChildSpanFromContext(span_context, "bft_client_request");
+  span.setTag("rid", config_.getreplicaId());
+  span.setTag("cid", m->getCid());
+  span.setTag("seq_num", reqSeqNum);
+
+  // A read only replica can handle only reconfiguration requests. Those requests are signed by the operator and
+  // the validation is done in the reconfiguration engine. Thus, we don't need to check the client validity as in
+  // the committers
+
+  if (reconfig_flag) {
+    LOG_INFO(GL, "ro replica has received a reconfiguration reqeust");
+    executeReadOnlyRequest(span, m);
+    delete m;
+    return;
+  }
+
+  delete m;
+}
+
+void ReadOnlyReplica::executeReadOnlyRequest(concordUtils::SpanWrapper &parent_span, ClientRequestMsg *request) {
```

Comment:
Nit: I think we can use a `const ClientRequestMsg&` type as we know it can't be nullptr and we don't change it. Speaking of which, we might also change:

```
char* ClientRequestMsg::requestBuf() const;
```
to return `const char*`.